vaadin-map-flow-parent/vaadin-map-flow-integration-tests/src/test/java/com/vaadin/flow/components/map/FeatureLabelIT.java

```diff
@@ -58,13 +68,64 @@ public void updateLabels() {
 
     @Test
     public void removeLabels() {
-        removeLabelText.click();
+        removeLabelButton.click();
 
         Assert.assertNull(getMarkerTextStyle("marker1").getText());
         Assert.assertNull(getMarkerTextStyle("marker2").getText());
         Assert.assertNull(getMarkerTextStyle("marker3").getText());
     }
 
+    @Test
+    public void setSetLabelStyle() {
+        setLabelStyleButton.click();
+
+        MapElement.TextReference text = getMarkerTextStyle("marker1");
+        Assert.assertEquals("Marker label 1", text.getText());
+        Assert.assertEquals("bold 13px monospace", text.getFont());
+        Assert.assertEquals(30, text.getOffsetX());
+        Assert.assertEquals(0, text.getOffsetY());
+        Assert.assertEquals("left", text.getTextAlign());
+        Assert.assertEquals("bottom", text.getTextBaseline());
+        Assert.assertEquals("#fff", text.getFill().getColor());
+        Assert.assertEquals("#000", text.getStroke().getColor());
+        Assert.assertEquals(5, text.getStroke().getWidth());
+        Assert.assertEquals("#1F6B75", text.getBackgroundFill().getColor());
+        Assert.assertEquals("#fff", text.getBackgroundStroke().getColor());
+        Assert.assertEquals(2, text.getBackgroundStroke().getWidth());
+
+        waitUntil(driver -> getRenderCount() == 1);
```

Comment:
All the style assertions can only check that the style function returns correct values, but not that markers have actually been re-rendered using the new styles. 

Adding a check for the number of renders should improve this a bit, especially since changing any style in OpenLayers requires manually forcing a re-render.